install.sh

```diff
@@ -198,6 +198,35 @@ if [ "$ZOOKEEPER_SNAPSHOT_FOLDER_EXISTS" -eq "1" ]; then
   fi
 fi
 
+# [begin] Snuba/Clickhouse transactions table rebuild
+clickhouse_query () { $dcr clickhouse clickhouse-client --host clickhouse -q "$1"; }
+$dc up -d clickhouse
+set +e
+CLICKHOUSE_CLIENT_MAX_RETRY=5
+# Wait until clickhouse server is up
+until clickhouse_query 'SELECT 1' > /dev/null; do
+  ((CLICKHOUSE_CLIENT_MAX_RETRY--))
+  [[ CLICKHOUSE_CLIENT_MAX_RETRY -eq 0 ]] && echo "Clickhouse server failed to come up in 5 tries." && exit 1;
+   echo "Trying again. Remaining tries #$CLICKHOUSE_CLIENT_MAX_RETRY"
+  sleep 0.5;
+done
+set -e
+
+SNUBA_HAS_TRANSACTIONS_TABLE=$(clickhouse_query 'EXISTS TABLE transactions_local' | tr -d '\n\r')
+SNUBA_TRANSACTIONS_NEEDS_UPDATE=$([ "$SNUBA_HAS_TRANSACTIONS_TABLE" == "1" ] && clickhouse_query 'SHOW CREATE TABLE transactions_local' | grep -v 'SAMPLE BY' || echo '')
+
+if [ "$SNUBA_TRANSACTIONS_NEEDS_UPDATE" ]; then
+  SNUBA_TRANSACTIONS_TABLE_CONTENTS=$(clickhouse_query "SELECT * FROM transactions_local LIMIT 1")
```

Comment:
I would just ensure these queries cannot individually transitorily fail silently. 
Like if this query fails temporarily, it is important that the failure is not treated in the same way as a "table empty" response since it may trigger a drop of the table by mistake.
Is a failure here going to stop the script ? (desirable)

Reply:
The whole script is in -e mode which makes it fail on non-zero exits. I only temporarily disable that during the clickhouse server up check. So no, if they error out, the script will exit.